traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1638,7 +1640,11 @@ func Update(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if server.XMPPID != nil && *server.XMPPID != originalXMPPID {
+	if server.XMPPID == nil || *server.XMPPID == "" {
+		log.Infof("Found empty XMPPID for server %s, setting random XMPPID", *server.HostName)
+		server.XMPPID = newUUID()
```

Comment:
We can't really do this either -- we'd have to set it to the hostname instead. Or rather, we shouldn't set it at all in an update, but just assume that nil/empty means hostname.

Reply:
Not setting XMPPID in an update in cb1b9976cc.